### PR TITLE
Build improvements

### PR DESCRIPTION
**Pull request by bryevdv**
Tryin to clean things up a bit

### CODE REVIEW

**Comment by bryevdv** [comment]
@mattpap what exactly is needed for `node make lint` ? it's also doing `build:all` but that runs the `generate_defaults.py` script which assumes some bokeh is installed. Would like to trim the command down to the bare minimum

**Comment by bryevdv** [comment]
It looks like there are some issues with the a released version of Bokeh getting installed instead of the local one. Additionally in another PR, noticed the examples test report a mismatch:

> [JS] {warning} file:///home/runner/miniconda3/lib/python3.7/site-packages/bokeh/server/static/js/bokeh.js:1105:34 [bokeh] Library versions: JS (2.0.2) / Python (145fe2c)

**Comment by bryevdv** [comment]
This is getting closer @mattpap any idea why the JS tests might fail in https://github.com/bokeh/bokeh/runs/707385131?check_suite_focus=true (or how to get more info)?

**Comment by mattpap** [comment]
> what exactly is needed for node make lint ? it's also doing build:all but that runs the generate_defaults.py script which assumes some bokeh is installed. Would like to trim the command down to the bare minimum

It's a codebase test, which implies that everything (lib, test, examples) must build correctly and lint. It's unfortunate that bokehjs' tests depend on Python, but right now it's too much work change that. This is also the reason why I keep both bokehjs-ci workflow and and js tests in bokeh-ci workflow. Optimally bokehjs would be self sufficient and contained in bokehjs-ci.

**Comment by mattpap** [comment]
> This is getting closer @mattpap any idea why the JS tests might fail in (...)

`node make examples` wasn't run (to compile bokehjs' examples).

**Comment by bryevdv** [comment]
> It's unfortunate that bokehjs' tests depend on Python, but right now it's too much work change that.

It's a cross-language test so utltimately it's got to go in one or the other, no?

**Comment by bryevdv** [comment]
@mattpap this can be deleted now, correct?
 
> UPLOAD="`[ -z $AWS_ACCESS_KEY_ID ] && echo '' || echo '--upload'`"

**Comment by mattpap** [comment]
> It's a cross-language test so utltimately it's got to go in one or the other, no?

Both bokeh and bokehjs could generate the "spec", which then could be compared independently. However, both require non-trivial setup to actually generate the spec, because both codes need to be executed. In particular bokehjs' requires a web browser and all the associated setup for that.

**Comment by mattpap** [comment]
> this can be deleted now, correct?

Yes.

**Comment by bryevdv** [comment]
@mattpap Just FYI we are testing against the source tree, not the build package. Have been for a while / forever AFAICT. That's not the worst thing but also not ideal or intended. I do want to fix it, maybe even by renaming the `bokeh` dir to `src` but thats more than I can tackle in this PR, so just going to restore the old behavior for now.

**Comment by bryevdv** [comment]
All right, CI yaml file now < 500 lines 🎉 If only GH workflows supported job templates like azure devops it would probably be reduced another 80%

Anyway, not quite done but close. A few observations:

* jobs with the conda GH actions seem a little bit slower
* the 3.8 conda installs sometimes seemingly randomly faily

Currently, using the mode that installs a python version up front, then does a `conda env update -f` with a generic env yaml file. As much as I don't like it, I having separate python-versioned yaml env files might be both faster and more robust, so I will try that next.

**Comment by bryevdv** [comment]
So I had mistakenly turned on the flag to use only bz2 packages. It's faster without it and perhaps more stable. I might be able to avoid the split env yaml files at least for now

File: .github/workflows/bokeh-ci.yml [hunk 1]
```diff
@@ -139,76 +69,43 @@ jobs:
...
-      - name: Run bokehjs codebase checks
-        working-directory: ./bokehjs
-        shell: bash
-        run: |
-          node make build:all lint
+      # - name: Run bokehjs codebase checks
+      #   working-directory: ./bokehjs
+      #   shell: bash -l {0}
+      #   run: |
+      #     node make build:all lint
```
Comment (bryevdv) [inline .github/workflows/bokeh-ci.yml:85]: @mattpap what you do you want to do about this in the immediate term:

* make codebase depend on build?
* remove here / move to BokehJS-CI?
* something else?

**Comment by bryevdv** [comment]
not sure why random jobs are hanging on the GH side at random places, fairly frustrating.

File: .github/workflows/bokeh-ci.yml (hunk 1 above)
Comment (mattpap) [reply .github/workflows/bokeh-ci.yml:85]: I will answer tomorrow.

**Comment by bryevdv** [comment]
OK split env files seems both faster and more reliable. Also I suppose it affords the opportunity to customize things by python version should the need arise. They change infrequently enough that the burden of having them split is tolerable, and honestly the only real "danger" is that something becomes extra/unnecessary and is forgotten to be removed (if anything is *missing* we will know when the tests fails)

@mattpap if you let me know what you want to do about the js codebase tests, this is otherwise ready to merge.

**Comment by bryevdv** [comment]
OK, I am just going to merge this now and make a separate issue to  restore BokehJS linting (in one way or another)

**Pull request merged**